ee/clickhouse/queries/funnels/funnel_correlation_persons.py

```diff
@@ -1,144 +1,189 @@
-from typing import Any, Dict, List, Optional, Tuple, cast
+from typing import Any, Dict, List, Optional, Tuple, Union
 
+from django.db.models.query import QuerySet
 from rest_framework.exceptions import ValidationError
-from rest_framework.utils.serializer_helpers import ReturnDict, ReturnList
 
-from ee.clickhouse.client import sync_execute
-from ee.clickhouse.models.property import get_property_string_expr
-from ee.clickhouse.queries.column_optimizer import ColumnOptimizer
+from ee.clickhouse.models.property import prop_filter_json_extract
+from ee.clickhouse.queries.actor_base_query import ActorBaseQuery, SerializedGroup, SerializedPerson
 from ee.clickhouse.queries.funnels.funnel_correlation import FunnelCorrelation
 from ee.clickhouse.queries.funnels.funnel_event_query import FunnelEventQuery
+from ee.clickhouse.queries.groups_join_query import GroupsJoinQuery
 from ee.clickhouse.queries.person_query import ClickhousePersonQuery
-from ee.clickhouse.sql.person import GET_TEAM_PERSON_DISTINCT_IDS
 from posthog.constants import FUNNEL_CORRELATION_PERSON_LIMIT, FunnelCorrelationType
 from posthog.models import Person
 from posthog.models.entity import Entity
 from posthog.models.filters.filter import Filter
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.group import Group
 from posthog.models.team import Team
 
 
-class FunnelCorrelationPersons:
+class FunnelCorrelationActors:
     def __init__(self, filter: Filter, team: Team, base_uri: str = "/") -> None:
-        self._funnel_correlation = FunnelCorrelation(filter, team, base_uri=base_uri)
+        self._base_uri = base_uri
         self._filter = filter
         self._team = team
 
         if not self._filter.correlation_person_limit:
             self._filter = self._filter.with_data({FUNNEL_CORRELATION_PERSON_LIMIT: 100})
 
-    def run(self):
-        """
-        Returns `ReturnList` type, generated by `serializers.serialize`, which returns the Person model.
-        """
+    def get_actors(
+        self,
+    ) -> Tuple[Union[QuerySet[Person], QuerySet[Group]], Union[List[SerializedGroup], List[SerializedPerson]]]:
```

Comment:
Nit: This return type is quite complex. There's IMO two clearer ways to write this:

```
Union[Tuple[QuerySet[Person], List[SerializedPerson]], Tuple[QuerySet[Group],List[SerializedGroup]]
```

Or using union types for Person | Group, SerializedPerson | SerializedGroup